services/publicapi/run_transaction.go

```diff
@@ -0,0 +1,50 @@
+package publicapi
```

Comment:
why did you name the file `run_transaction`?

Reply:
because i'm advocating changing callmethod to RunReadOnlyTransaction so that these three functions will sound the same.